teps/0070-tekton-catalog-task-platform-support.md

```diff
@@ -0,0 +1,137 @@
+---
+status: proposed
+title: Platform support in Tekton catalog
+creation-date: '2021-06-02'
+last-updated: '2021-06-02'
+authors:
+- '@barthy1'
+---
+
+# TEP-0070: Platform support in Tekton catalog
+
+- [Summary](#summary)
+- [Motivation](#motivation)
+  - [Goals](#goals)
+  - [Non-Goals](#non-goals)
+  - [Use Cases (optional)](#use-cases-optional)
+- [Requirements](#requirements)
+- [Proposal](#proposal)
+  - [Catalog side](#catalog-side)
+  - [Hub side](#hub-side)
+  - [Tekton CLI side](#tekton-cli-side)
+- [References (optional)](#references-optional)
+
+## Summary
+
+This TEP proposes to add information about supported operating systems/hardware architectures (platforms) to the Tekton catalog resources and provide ability to the user to filter/search resources from the catalog for specific platform.
+
+## Motivation
+
+Tekton at this moment has releases for 4 hardware architectures(platforms) - `linux/amd64`, `linux/s390x`, `linux/ppc64le` and `linux/arm64`. Tekton catalog contains already great list of resources to run. However there is no information about which resource can be executed on which platform. As a result user can fail running a task on top of not supported hardware architecture.
+
+### Goals
+
+- Define the way to add supported platforms into resource definition in the Tekton catalog.
+- Define the ways to show information about platforms to the user.
+
+### Non-Goals
+
+- Define modifications in Tekton to better support running tasks on different platforms ([TEP-0041](https://github.com/tektoncd/community/pull/310)).
+- Limit/hide ("no-show" or "no-access") catalog resources based on host platform.
```

Comment:
@afrittoli I thought it could make sense to separate how tasks in catalog are marked as executable for different platforms from how tekton actually runs them for different platforms. 

> I think that the design of how we add the list of supported platforms depends on how that information is going to be consumed, including the tekton controller.

Do you suggest to have some implemented logic which based `tekton.dev/platforms` annotation actually runs task differently? I am not sure that it's doable at this moment, as there are too many differences, which specific modifications can be required(or not required) for different platforms (however I can be just too pessimistic..)

> Also not covered by this TEP but probably related is the support for multiple OSes. With multiple OSes we would probably end-up having to handle multiple (OS-specific) versions of tasks, which hopefully won't be the case for different architectures.

Again if consider this TEP only in area of marking tasks with platform information, the idea you mentioned is out of scope ;) However I can imagine that tasks in catalog with the same name, but different `tekton.dev/platforms` can be provided to the final user as one task.